pandas/core/tools/datetimes.py

```diff
@@ -765,6 +765,7 @@ def to_datetime(
         - list-like: DatetimeIndex
         - Series: Series of datetime64 dtype
         - scalar: Timestamp
+        - list-like with mixed timezones: Index of object dtype
```

Comment:
Actually on second thought might be better to combine with `list-like: DatetimeIndex` like

```
- Index:
    - DatetimeIndex if timezone naive or aware with the same timezone
    - Index of object dtype if timezone aware with mixed timezones
```